matlab/src/Communicator.cpp

```diff
@@ -59,15 +59,7 @@ Ice_Communicator_stringToProxy(void* self, const char* s, void** proxy)
 {
     try
     {
-        auto p = deref<Ice::Communicator>(self)->stringToProxy(s);
-        if(p)
-        {
-            *proxy = createShared<Ice::ObjectPrx>(p);
-        }
-        else
-        {
-            *proxy = 0;
-        }
+        *proxy = createProxy(deref<Ice::Communicator>(self)->stringToProxy(s));
```

Comment:
I added two overloads for createProxy:
 - the ObjectPrx overload never returns nullptr
 - the std::optional<ObjectPrx> overload returns a nullptr when the arg is nullopt